### PR TITLE
Add option to select connection 

### PR DESCRIPTION
**Pull request by PedroFelli**
#23 
- [x] Add Manage Database on side menu
- [x] Connect to database using credentials 

https://github.com/user-attachments/assets/63ae1817-ce4f-483e-b6d6-f028fc2e8ab4

### CODE REVIEW

File: src/app/(main)/query-runner/_components/cql-editor.tsx [hunk 1]
```diff
@@ -96,9 +96,17 @@ export function CqlEditor() {
 			fetchSizeRef.current === "unset" || !fetchSizeRef.current
 				? undefined
 				: Number.parseInt(fetchSizeRef.current);
+
+		const { username, password, ...rest } = currentConnectionRef.current;
+		const connection = {
+			...rest,
+			username: username ?? null,
+			password: password ?? null,
```
Comment (Daniel-Boll) [inline src/app/(main)/query-runner/_components/cql-editor.tsx:104]:
```suggestion
			username,
			password
```

This `Nullish coalescing` being used this way doesn't make much sense, if it's null you'll be attributing null to it again.

**Review by Daniel-Boll — CHANGES_REQUESTED** [review]

File: src/app/(main)/query-runner/_components/cql-editor.tsx (hunk 1 above)
Comment (PedroFelli) [reply src/app/(main)/query-runner/_components/cql-editor.tsx:104]: > This `Nullish coalescing` being used this way doesn't make much sense, if it's null you'll be attributing null to it again.

You're right, using nullish coalescing like that didn't make sense. I removed it and instead added auth conditionally, only if both username and password are present. This way, if either is null, auth won't be included at all.

File: src/actions/query-keyspaces.ts [hunk 2]
```diff
@@ -3,23 +3,38 @@
...
+        };
+
+        if (username && password) {
+            clusterConfig.auth = {
+                username,
+                password,
+            };
+        }
+
+        const session = await new Cluster(clusterConfig).connect();
```
Comment (Daniel-Boll) [inline src/actions/query-keyspaces.ts:34]: That's not the best idea, here you are giving up the reference to the Cluster Class itself, we don't know how this is going to be garbage collected in the server as the client action lifespan is abstract away, the best approach here is to actually keep as it was before.

File: src/actions/query-keyspaces.ts [hunk 3]
```diff
@@ -3,23 +3,38 @@
...
+        const clusterConfig: any = {
+            nodes: [`${host}:${port}`],
+        };
+
+        if (username && password) {
+            clusterConfig.auth = {
+                username,
+                password,
+            };
+        }
```
Comment (Daniel-Boll) [inline src/actions/query-keyspaces.ts:32]: You might as well check the [src/actions/execute-query.ts](https://github.com/basementdevs/scylla-studio/pull/22/files#diff-a624c47dd8ce7ac4b51e54c369ef4af92cf8b3e24a8b45c29ae1e594eb707f81) file to see how the auth is being established and do roughly the same.

Comment (PedroFelli) [reply src/actions/query-keyspaces.ts]: ok

**Review by Daniel-Boll — APPROVED** [review]
Perfect, thanks for your contribution @PedroFelli! This PR is hacktoberfest compliant, you can sync your GitHub account in https://hacktoberfest.com/ and track it from there.

Best regards,
Daniel Boll. 🎴

**Pull request merged**